sdks/python/apache_beam/runners/runner.py

```diff
@@ -168,6 +188,7 @@ def visit_transform(self, transform_node):
           raise
 
     pipeline.visit(group_by_key_input_visitor())
+    pipeline.visit(flatten_input_visitor())
```

Comment:
Please add a comment here as to why. Similarly with the group_by_key_input_visitor.